### PR TITLE
Firewalld update -- exposed more fields, updated docs, added custom matchers

### PR DESCRIPTION
**Pull request by wdower**
## Description

We needed to retrieve the target of an active firewall zone and noticed that the target and some other properties were missing from the firewalld resource. So We added these 9 properties:

target
icmp_block_inversion
ports
protocols
masquerade
forward_ports
source_ports
icmp_blocks
rich_rules

We also created mock tests for all the new properties. Currently updating the docs.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ x ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by dkumaras** [comment]
Pls chk & let us know if the edits are fine.

File: docs-chef-io/content/inspec/resources/firewalld.md [hunk 1]
```diff
@@ -40,10 +40,24 @@ Use the where clause to test open interfaces, sources, and services in active zo
...
+- `services`
+- `target`
+- `ports`
+- `protocols`
+- `forward_ports`
+- `source_ports`
+- `icmp_blocks`
+- `rich_rules`
+
+## Property Examples
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/firewalld.md]: We are removing the `Property Examples` section from the docs . We are going to list out all `Property Examples` under the same head `Property`. So doing that change will help reduce the rework.

**Comment by Vasu1105** [comment]
@dkumaras The docs changes for indentation breaks formatting that needs to be corrected.

**Comment by aaronlippold** [comment]
@wdower @dtingstrom any progress on getting these issues resolved to move the PR out of draft?

File: docs-chef-io/content/inspec/resources/firewalld.md (hunk 1 above)
Comment (wdower) [reply docs-chef-io/content/inspec/resources/firewalld.md]: @Vasu1105 Removed the old `Properties` section and renamed `Property Examples` as the new `Properties` -- let me know if that works for how you want the docs to look.

**Comment by wdower** [comment]
@Vasu1105 any work I still need to do on this PR?

**Review by clintoncwolfe — APPROVED** [review]
This looks good, thanks!

**Pull request merged**